saleor/checkout/utils.py

```diff
@@ -1,27 +1,619 @@
+    user_form = AnonymousUserBillingForm(data, instance=cart)
+
+    updated = False
+
+    if user_form.is_valid() and address_form.is_valid() and not preview:
+        user_form.save()
+        address = address_form.save()
+        change_billing_address_in_cart(cart, address)
+        updated = True
+
+    return user_form, address_form, updated
+
+
+def get_summary_without_shipping_forms(cart, user_addresses, data, country):
+    """Forms initialized with data depending on addresses in cart."""
+    billing_address = cart.billing_address
+
+    if billing_address and billing_address in user_addresses:
+        address_form, preview = get_address_form(
+            data,
+            autocomplete_type='billing',
+            country_code=billing_address.country.code,
+            initial={'country': billing_address.country})
+        initial_address = billing_address.id
+    elif billing_address:
+        address_form, preview = get_address_form(
+            data,
+            autocomplete_type='billing',
+            country_code=billing_address.country.code,
+            initial={'country': billing_address.country},
+            instance=billing_address)
+        initial_address = AddressChoiceForm.NEW_ADDRESS
+    else:
+        address_form, preview = get_address_form(
+            data,
+            autocomplete_type='billing',
+            country_code=country.code,
+            initial={'country': country})
+        if cart.user and cart.user.default_billing_address:
+            initial_address = cart.user.default_billing_address.id
+        else:
+            initial_address = AddressChoiceForm.NEW_ADDRESS
+
+    addresses_form = AddressChoiceForm(
+        data, addresses=user_addresses, initial={'address': initial_address})
+    return address_form, addresses_form, preview
+
+
+def update_billing_address_in_cart(cart, user_addresses, data, country):
+    """Return shipping address choice forms and if an address was updated."""
+    address_form, addresses_form, preview = (
+        get_summary_without_shipping_forms(
+            cart, user_addresses, data, country))
+
+    updated = False
+
+    if addresses_form.is_valid():
+        use_existing_address = (
+            addresses_form.cleaned_data['address'] !=
+            AddressChoiceForm.NEW_ADDRESS)
+
+        if use_existing_address:
+            address_id = addresses_form.cleaned_data['address']
+            address = Address.objects.get(id=address_id)
+            change_billing_address_in_cart(cart, address)
+            updated = True
+
+        elif address_form.is_valid():
+            address = address_form.save()
+            change_billing_address_in_cart(cart, address)
+            updated = True
+
+    return addresses_form, address_form, updated
+
+
+def change_billing_address_in_cart(cart, address):
+    """Save billing address in cart if changed.
+
+    Remove previously saved address if not connected to any user.
+    """
+    has_address_changed = (
```

Comment:
can we introduce some kind of variables, or use `all`?  We are checking presence of 7 variables here

Reply:
As above, I moved it to a new function.